modules/uhttp/client/client.go

```diff
@@ -25,19 +25,83 @@ import (
 	"time"
 
 	"go.uber.org/fx/auth"
+	"go.uber.org/fx/dig"
+	"go.uber.org/fx/modules"
+	"go.uber.org/fx/service"
+
+	"github.com/opentracing/opentracing-go"
+)
+
+const (
+	_middlewareKey = "httpClientMiddleware"
+	_graphKey      = "httpClientGraph"
 )
 
+// WithOutbound lets you to add a custom middleware to the client
+func WithOutbound(middleware ...OutboundMiddleware) modules.Option {
+	return func(info *service.ModuleCreateInfo) error {
+		items := info.Items
+		if m, ok := items[_middlewareKey]; ok {
+			items[_middlewareKey] = append(m.([]OutboundMiddleware), middleware...)
+		} else {
+			items[_middlewareKey] = middleware
+		}
+		return nil
+	}
+}
+
+// WithGraph allows you to use a custom dependency injection graph to resolve Tracer and AuthInfo
+func WithGraph(graph dig.Graph) modules.Option {
```

Comment:
And they will not have to know, a regular `client.New()` will return a client, that is initialized with a default graph.

They may need to override a tracer/auth client, then they can provide their DI graph, that's why it is public. No one forces them to use it.